src/auth.rs

```diff
@@ -61,30 +61,34 @@
 //! ```
 //! [`SecretKey`]: https://docs.rs/orion/latest/orion/auth/struct.SecretKey.html
 //! [`SecretKey::default()`]: https://docs.rs/orion/latest/orion/auth/struct.SecretKey.html
-//! [`Tag`]: https://docs.rs/orion/latest/orion/hazardous/mac/hmac/struct.Tag.html
+//! [`Tag`]: struct.Tag.html
 
-use crate::{errors::UnknownCryptoError, hazardous::mac::hmac};
-pub use crate::{hazardous::mac::hmac::Tag, hltypes::SecretKey};
+pub use crate::hltypes::{Blake2bTag as Tag, SecretKey};
+use crate::{errors::UnknownCryptoError, hazardous::hash::blake2b};
+
+/// The recommended Tag size (bytes) to be output by Blake2b in keyed mode.
+const BLAKE2B_TAG_SIZE: usize = 16;
 
 #[must_use]
-/// Authenticate a message using HMAC-SHA512.
+/// Authenticate a message using Blake2b in keyed mode.
 pub fn authenticate(secret_key: &SecretKey, data: &[u8]) -> Result<Tag, UnknownCryptoError> {
-	let mut state = hmac::init(&hmac::SecretKey::from_slice(
-		secret_key.unprotected_as_bytes(),
-	)?);
+	let blake2b_secret_key = blake2b::SecretKey::from_slice(secret_key.unprotected_as_bytes())?;
+	let mut state = blake2b::init(Some(&blake2b_secret_key), BLAKE2B_TAG_SIZE)?;
 	state.update(data)?;
-	state.finalize()
+	let blake2b_digest = state.finalize()?;
+	Tag::from_slice(blake2b_digest.as_ref())
 }
 
 #[must_use]
-/// Authenticate and verify a message using HMAC-SHA512.
+/// Authenticate and verify a message using Blake2b in keyed mode.
 pub fn authenticate_verify(
 	expected: &Tag,
 	secret_key: &SecretKey,
 	data: &[u8],
 ) -> Result<bool, UnknownCryptoError> {
-	let key = hmac::SecretKey::from_slice(secret_key.unprotected_as_bytes())?;
-	hmac::verify(expected, &key, data)
```

Comment:
The same 32-byte minimum limitation on SecretKey should be checked here, as well as tested.

Reply:
Oh, I thought that the internal call to `blake2b::SecretKey::from_slice` did the check automatically. But that doesn't do any checks. So we have to do one in this function. Does that sound right?

It also seems like we should return an `UnknownCryptoError` instead of panicking, right?

Reply:
Exactly, but both in `authenticate()` and `authenticate_verify()`. 

Note: `blake2b::SecretKey::from_slice` does do a check, but checks that the slice is in range of `1..=64`.